qiskit_dynamics/signals/signals.py

```diff
@@ -573,7 +547,7 @@ def __init__(self, *signals, name: Optional[str] = None):
             elif isinstance(sig, Signal):
                 components.append(sig)
             elif isinstance(sig, (int, float, complex)) or (
-                isinstance(sig, Array) and sig.ndim == 0
+                not isinstance(sig, (int, float, complex)) and sig.ndim == 0
```

Comment:
Maybe this whole `elif` can be changed to:
```
else:
    try:
        if unp.asarray(sig).ndim == 0:
            components.append(Signal(sig))
    except:
```

This way the final resort is to check if it can be converted to an array or not. If not, the error at the end will be raised. Perhaps the error message at the end should also be changed to:
```
"Components of a SignalSum must be instances of a Signal subclass or a scalar."
```

Reply:
I fixed at https://github.com/Qiskit-Extensions/qiskit-dynamics/pull/269/commits/2e1436e838dec1acbb5dd478b39a773d353741b4.